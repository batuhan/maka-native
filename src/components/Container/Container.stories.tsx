// src/components/Container/Container.stories.tsx

import React from "react";
import { Text } from "react-native";
import { storiesOf } from "@storybook/react-native";

import Container from "./Container";

storiesOf("Container", module).add("Container", () => (
  <Container>
    <Text>Container</Text>
  </Container>
));
