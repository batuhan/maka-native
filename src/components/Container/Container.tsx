import React, { ReactNode } from "react";
import { View } from "react-native";
import styled from "styled-components";

const ContainerView = styled(View)`
  background-color: ${(props: { bgColor?: string }) =>
    props.bgColor || "white"};
`;

interface Props {
  children: ReactNode;
  bgColor?: string | undefined;
}

export default function({ children, bgColor }: Props) {
  return <ContainerView bgColor={bgColor}>{children}</ContainerView>;
}
