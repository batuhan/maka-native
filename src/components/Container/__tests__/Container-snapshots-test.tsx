import React from "react";
import renderer from "react-test-renderer";

import "jest-styled-components";

import Container from "../Container";
import { Text } from "react-native";

describe("Container", () => {
  const children = <Text>Container Children</Text>;
  
  describe("Rendering", () => {
    it("should match to snapshot", () => {
      const tree = renderer.create(<Container>{children}</Container>).toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule("background-color", "white");
    });

    it("should match to snapshot", () => {
      const bgColor = "black";
      const tree = renderer
        .create(<Container bgColor={bgColor}>{children}</Container>)
        .toJSON();
      expect(tree).toHaveStyleRule("background-color", "black");
    });
  });
});
