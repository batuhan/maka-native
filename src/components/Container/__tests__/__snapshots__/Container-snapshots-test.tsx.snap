// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Rendering should match to snapshot 1`] = `
.c0 {
  background-color: white;
}

<View
  className="c0"
>
  <Text>
    Container Children
  </Text>
</View>
`;
